src/commonMain/kotlin/com.adamratzman.spotify/models/SpotifyUris.kt

```diff
@@ -29,20 +25,14 @@ private fun String.remove(type: String): String {
     throw SpotifyUriException("Illegal Spotify ID/URI: '$this' isn't convertible to '$type' id")
 }
 
-private class SimpleUriSerializer<T : SpotifyUri>(val ctor: (String) -> T) : KSerializer<T> {
-    override val descriptor: SerialDescriptor = StringDescriptor
-    override fun deserialize(decoder: Decoder): T = ctor(decoder.decodeString())
-    override fun serialize(encoder: Encoder, obj: T) = encoder.encodeString(obj.uri)
-}
-
 /**
  * Represents a Spotify URI, parsed from either a Spotify ID or taken from an endpoint.
  *
  * @property uri retrieve this URI as a string
  * @property id representation of this uri as an id
  */
 @Serializable
-sealed class SpotifyUri(val input: String, type: String) {
+sealed class SpotifyUri(val input: String, val type: String) {
```

Comment:
Why is `type` public? This shouldn't be used, as it's way more safe to rely on the type system aka the compiler, instead of the logic you need to write when handling strings.